mbq/client/client.py

```diff
@@ -11,12 +10,17 @@
 
 class ServiceClient:
     def __init__(self, api_url, auth=None, headers=None, post_process_response=None,
-                 default_timeout=30):
+                 correlation_id_getter=None, default_timeout=30):
+        """ We use correlation_ids to track the flow of a request between different services.
+        It is the responsibility of users of ServiceClient to implement a correlation_id_getter
+        which either passes along or provides a new cid when making an outbound request.
```

Comment:
Generating a new CID _at the time of making an outbound request_ will pretty much always be wrong, I wouldn't mention it in the docstring.